### ISSUE TITLE
Bypass checking URLs in comments for 'max-line-length' rule

### ISSUE THREAD

**Issue by roman-petrov** [palantir/tslint#2693]
The case is we have some long URLs in comments in our code base . These URLs sometimes are quite long, so we have to use `tslint:disable-next-line:max-line-length` before long URLs. My feature request/suggestion is to modify `max-line-length` rule to allow long URLs in comments, or add an option to allow such URLs.

**Comment by ajafff** [comment]
There was a similar request to allow long lines for import statements. IIRC the resolution was that tslint won't fix this and you should just disable the rule for that line. 

Personally I'd like to add some options to the rule, too. 

In the mean you can use `max-len` from `tslint-eslint-rules`

**Comment by roman-petrov** [comment]
Just a suggestion: maybe it makes sense to just add an option like `ignoreRegexp` which will contain array of regular expressions?..

**Comment by bennyn** [comment]
@roman-petrov, we have the same issue with our code. To give an example:

```typescript
/**
 * Definition of OAuth 2.0 Scopes for Google Plus.
 * @see https://developers.google.com/resources/api-libraries/documentation/plus/v1/java/latest/com/google/api/services/plus/PlusScopes.html
 */
export default class PlusScopes {
  // Know the list of people in your circles, your age range, and language
  public static PLUS_LOGIN = 'https://www.googleapis.com/auth/plus.login';
  // Know who you are on Google
  public static PLUS_ME = 'https://www.googleapis.com/auth/plus.me';
  // View your email address
  public static USERINFO_EMAIL = 'https://www.googleapis.com/auth/userinfo.email';
  // View your basic profile info
  public static USERINFO_PROFILE = 'https://www.googleapis.com/auth/userinfo.profile';
}
```

The URL next to `@see` is too long when using `"extends": "tslint:recommended"` in `tslint.json`.

In ESLint there is a [ignoreComments](http://eslint.org/docs/rules/max-len#ignorecomments) option which lets you bypass long comments. Would be nice if TSLint had something similar.

**Comment by adidahiya** [comment]
We'd take a PR for a rule option that exempts URLs from the rule.

**Comment by xjose97x** [comment]
@adidahiya  please also long regular expressions.

**Comment by adidahiya** [comment]
@xjose97x I'm reluctant to add regular expressions for now because of the performance impact. Can you describe your use cases for regex?

**Comment by xjose97x** [comment]
@adidahiya I have a regex for checking email addresses:
/^(([^<>()[\]\\.,;:\s@\"]+(\.[^<>()[\]\\.,;:\s@\"]+)*)|(\".+\"))@((\[[0-9]{1,3}\.[0-9]{1,3}\.[0-9]{1,3}\.[0-9]{1,3}\])|(([a-zA-Z\-0-9]+\.)+[a-zA-Z]{2,}))$/

Which makes tslint bring up the max-line-length error. I actually like that tslint rule, but it should be disabled for cases like regex and urls.

**Comment by adidahiya** [comment]
ah, sorry, I thought you were asking for a rule option that takes a regex to whitelist certain kinds of text from the `max-line-length` rule. exempting regex syntax itself is much simpler; we can also take a PR for that

**Comment by myrdd** [comment]
I am using the following pattern to ignore long urls. I allow one to three ”other“ characters at the beginning to allow for listings (`- http://....`).

    "max-line-length": [
      true,
      {
        "ignore-pattern": "^\\s*// *(.{1,3} )?(https?)://[^ ]+$",
        "limit": 80
      }
    ]

**Comment by remi2j** [comment]
Similarly to @myrdd, I'm ignoring any line that contains 'http'

```
"max-line-length": [
  true,
  {
    "ignore-pattern": "http",
    "limit": 80
  }
]

**Comment by JoshuaKGoldberg** [comment]
☠️ *TSLint's time has come!* ☠️ 

TSLint is no longer accepting most feature requests per #4534. See **[typescript-eslint.io](https://typescript-eslint.io)** for the new, shiny way to lint your TypeScript code with **ESLint**. ✨ 

It was a pleasure open sourcing with you all!
